### ISSUE TITLE
[Bug] Node start-up is considerably slower after the snarkVM 0.7.5 update

### ISSUE THREAD

**Issue by ljedrz** [AleoHQ/snarkOS#1003]
When a node is started with a branch containing https://github.com/AleoHQ/snarkOS/pull/937, some initial process seems to be much slower than with a branch right before those changes.

**Comment by niklaslong** [comment]
Seems like `MerkleTree::new` called by `MerkleTreeLedger::<LedgerStorage>::open_at_path` (omitting a few levels of abstraction) in `main` is the culprit:
```
current staging: 298.77s
4fa859a7: 13.61s 
```
This is with `--release` and a db with a height of `333666`. 

Digging deeper 🔬.

**Comment by niklaslong** [comment]
I've managed to pinpoint the issue to `MerkleTree::hash_row` calling the `MerkleTree::hash_leaf` method in its loop. Comparison showing the time for each full execution of the inner loop and the length of the resulting collection:

SnarkVM `0.7.5`, snarkOS commit: `e056f04`. 
```
ELAPSED: 124.69770675s
LEN: 1356
ELAPSED: 119.94777025s
LEN: 1049
ELAPSED: 68.599668416s
LEN: 525
ELAPSED: 36.558126875s
LEN: 263
ELAPSED: 19.963686333s
LEN: 132
ELAPSED: 9.143612708s
LEN: 66
ELAPSED: 4.474370125s
LEN: 33
ELAPSED: 2.49359925s
LEN: 17
ELAPSED: 1.294850791s
LEN: 9
ELAPSED: 615.174916ms
LEN: 5
ELAPSED: 354.189083ms
LEN: 3
ELAPSED: 307.222208ms
LEN: 2
ELAPSED: 148.275208ms
LEN: 1
ELAPSED: 66.824583ms
LEN: 1
ELAPSED: 33.946958ms
LEN: 1
ELAPSED: 15.264583ms
LEN: 1
ELAPSED: 9.175083ms
LEN: 1
ELAPSED: 3.930916ms
LEN: 1
ELAPSED: 1.753ms
LEN: 1
ELAPSED: 864.375µs
LEN: 1
ELAPSED: 440.833µs
LEN: 1
```




SnarkVM `0.5.4`, snarkOS commit:`4fa859a7`. 
```
ELAPSED: 8.5022205s
LEN: 1356
ELAPSED: 5.962819875s
LEN: 1049
ELAPSED: 3.760678333s
LEN: 525
ELAPSED: 1.681926083s
LEN: 263
ELAPSED: 1.208915291s
LEN: 132
ELAPSED: 444.34775ms
LEN: 66
ELAPSED: 191.395416ms
LEN: 33
ELAPSED: 109.0415ms
LEN: 17
ELAPSED: 59.442958ms
LEN: 9
ELAPSED: 37.789125ms
LEN: 5
ELAPSED: 35.137708ms
LEN: 3
ELAPSED: 32.639208ms
LEN: 2
ELAPSED: 14.753833ms
LEN: 1
ELAPSED: 7.399916ms
LEN: 1
ELAPSED: 3.69225ms
LEN: 1
ELAPSED: 1.8595ms
LEN: 1
ELAPSED: 946.333µs
LEN: 1
ELAPSED: 479.583µs
LEN: 1
ELAPSED: 244.958µs
LEN: 1
ELAPSED: 147.958µs
LEN: 1
ELAPSED: 65.25µs
LEN: 1
```

Not yet sure if this is due to parameter changes.

**Comment by ljedrz** [comment]
This slowdown still occurs with `0.7.6`.

**Comment by ljedrz** [comment]
Also of note: this slowdown will be halved with the upcoming https://github.com/AleoHQ/snarkOS/pull/972, as it doesn't use the secondary storage anymore.

**Comment by ljedrz** [comment]
This is fixed in snarkVM `0.7.7`; keeping the issue open until snarkOS updates to it.

**Comment by Protryon** [comment]
@ljedrz snarkVM has been updated to `0.7.8`, good to close?
